x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/fields/delete_field_provider.tsx

```diff
@@ -78,27 +82,53 @@ export const DeleteFieldProvider = ({ children }: Props) => {
           onCancel={closeModal}
           onConfirm={confirmDelete}
           cancelButtonText={i18n.translate(
-            'xpack.idxMgmt.mappingsEditor.deleteFieldCancelButtonLabel',
+            'xpack.idxMgmt.mappingsEditor.deleteField.confirmationModal.cancelButtonLabel',
             {
               defaultMessage: 'Cancel',
             }
           )}
           buttonColor="danger"
           confirmButtonText={i18n.translate(
-            'xpack.idxMgmt.mappingsEditor.deleteFieldRemoveButtonLabel',
+            'xpack.idxMgmt.mappingsEditor.deleteField.confirmationModal.removeButtonLabel',
             {
               defaultMessage: 'Remove',
             }
           )}
         >
-          <Fragment>
-            <p>
-              {i18n.translate('xpack.idxMgmt.mappingsEditor.deleteDescription', {
-                defaultMessage: 'This will also delete the following fields.',
-              })}
-            </p>
-            <FieldsTree fields={fieldsTree} />
-          </Fragment>
+          <>
```

Comment:
I see the same logic is being used in `update_field_provider`, what do you think about creating something shared?